xarray/core/indexes.py

```diff
@@ -642,6 +656,8 @@ def from_variables(
 
         obj = cls(data, dim, coord_dtype=var.dtype)
         assert not isinstance(obj.index, pd.MultiIndex)
+        # Rename safely
```

Comment:
Can we duplicate your nice comment from above here too please?

Reply:
yes